crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -493,71 +567,78 @@ impl SystemStage {
-        if !(parallel.is_empty()
-            && at_start.is_empty()
-            && before_commands.is_empty()
-            && at_end.is_empty())
-        {
-            let mut string = "Execution order ambiguities detected, you might want to \
-                    add an explicit dependency relation between some of these systems:\n"
-                .to_owned();
-            if !parallel.is_empty() {
-                writeln!(string, " * Parallel systems:").unwrap();
-                write_display_names_of_pairs(&mut string, &self.parallel, parallel, world);
-            }
-            if !at_start.is_empty() {
-                writeln!(string, " * Exclusive systems at start of stage:").unwrap();
-                write_display_names_of_pairs(
-                    &mut string,
-                    &self.exclusive_at_start,
-                    at_start,
-                    world,
+        ) -> usize {
+            for (i, (system_a_index, system_b_index, conflicting_indexes)) in
+                ambiguities.iter().enumerate()
+            {
+                let _system_a_name = systems[*system_a_index].name();
+                let _system_b_name = systems[*system_b_index].name();
+
+                let _conflicting_components = conflicting_indexes
+                    .iter()
+                    .map(|id| world.components().get_info(*id).unwrap().name())
+                    .collect::<Vec<_>>();
+
+                let _ambiguity_number = i + offset;
+
+                println!(
+                    "{_ambiguity_number}. {_system_a_name} conflicts with {_system_b_name} on {_conflicting_components:?}"
                 );
             }
-            if !before_commands.is_empty() {
-                writeln!(string, " * Exclusive systems before commands of stage:").unwrap();
-                write_display_names_of_pairs(
-                    &mut string,
+
+            // We can't merge the SystemContainer arrays, so instead we manually keep track of how high we've counted :upside_down:
+            ambiguities.len()
+        }
+
+        let parallel = find_ambiguities(&self.parallel, &ambiguity_report.ignore_crates);
+        let at_start = find_ambiguities(&self.exclusive_at_start, &ambiguity_report.ignore_crates);
+        let before_commands = find_ambiguities(
+            &self.exclusive_before_commands,
+            &ambiguity_report.ignore_crates,
+        );
+        let at_end = find_ambiguities(&self.exclusive_at_end, &ambiguity_report.ignore_crates);
+
+        let mut unresolved_count = parallel.len();
+        unresolved_count += at_start.len();
+        unresolved_count += before_commands.len();
+        unresolved_count += at_end.len();
+
+        if unresolved_count > 0 {
+            println!("{unresolved_count} unresolved system order ambiguities detected in one of your stages: you might want to \
+            add an explicit system ordering between some of these systems to avoid logic errors. \n");
```

Comment:
```suggestion
            println!("One of your stages contains {unresolved_count} pairs of systems with unknown order and conflicting data access. \
            You may want to add `.before()` or `.after()` constraints between some of these systems to prevent bugs.\n");
```

Reply:
I've been wanting to simplify the language in these warning messages for a while. 

It'd be even better if you could include the name of the stage here.